src/libraries/System.Private.CoreLib/src/System/Half.cs

```diff
@@ -1578,9 +1578,11 @@ public static int ILogB(Half x)
         public static Half Lerp(Half value1, Half value2, Half amount) => (Half)float.Lerp((float)value1, (float)value2, (float)amount);
 
         /// <inheritdoc cref="IFloatingPointIeee754{TSelf}.ReciprocalEstimate(TSelf)" />
+        [MethodImpl(MethodImplOptions.AggressiveOptimization)]
```

Comment:
It does not look right to spread this around like this. 

The component responsible for non-deterministic expansion is the JIT. Would it be best to deal in the JIT so that the both the cause and the mitigation are in the same place? 
- Do not perform the non-deterministic intrinsic expansion in the JIT for R2R
- Fail compilation of the self-recursive method implementation for methods with non-deterministic intrinsic expansion

Reply:
We need to allow for self-recursive expansion to happen during JIT time still, as it's necessary for delegates to these methods to be invokable (this is notably why the test is failing, its using such a delegate).

I think we can easily fail it for R2R, however.